include/yaramod/types/regexp.h

```diff
@@ -257,25 +313,35 @@ class RegexpOperation : public RegexpUnit
 
 	virtual std::string getText() const override
 	{
-		return _operand->getText() + _operation + (_greedy ? std::string() : "?");
+		assert(_operand != nullptr);
+		return _operand->getText() + _operation->getString() + _greedy->getPureText();
 	}
 
-	char getOperation() const { return _operation; }
+	virtual char getOperation() const { return _operation->getString()[0]; }
 
-	bool isGreedy() const { return _greedy; }
+	bool isGreedy() const { return _greedy->getBool(); }
 
 	const std::shared_ptr<RegexpUnit>& getOperand() const { return _operand; }
 
 	void setOperand(const std::shared_ptr<RegexpUnit>& operand) { _operand = operand; }
 	void setOperand(std::shared_ptr<RegexpUnit>&& operand) { _operand = std::move(operand); }
 
 protected:
-	RegexpOperation(char operation, std::shared_ptr<RegexpUnit>&& operand, bool greedy) : _operation(operation), _operand(std::move(operand)), _greedy(greedy) {}
+	RegexpOperation(TokenType operation_token_type, char operation_symbol, std::shared_ptr<RegexpUnit>&& operand, bool greedy)
+		: _operand(std::move(operand))
+	{
+		//take the operand's tokenStream and append first the operation and then greedy
+		_tokenStream = std::move(_operand->getTokenStream());
+		_operation = _tokenStream->emplace_back(operation_token_type, std::string() + operation_symbol);
```

Comment:
Same as before, if you are trying to construct string from char then use constructor (2) https://en.cppreference.com/w/cpp/string/basic_string/basic_string

Reply:
Fixed.